src/OrchardCore.Modules/OrchardCore.Lucene/Recipes/LuceneIndexRebuildStep.cs

```diff
@@ -19,24 +20,34 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                 return;
             }
 
-            var indices = context.Step["Indices"];
-            if (indices != null)
+            var model = context.Step.ToObject<LuceneIndexRebuildStepModel>();
+
+            if (model.IncludeAll || model.Indices.Length > 0)
             {
-                await HttpBackgroundJob.ExecuteAfterEndOfRequestAsync("lucene-index-rebuild", async (scope) => {
+                await HttpBackgroundJob.ExecuteAfterEndOfRequestAsync("lucene-index-rebuild", async (scope) =>
+                {
                     var luceneIndexSettingsService = scope.ServiceProvider.GetRequiredService<LuceneIndexSettingsService>();
                     var luceneIndexingService = scope.ServiceProvider.GetRequiredService<LuceneIndexingService>();
-                    foreach (var indexToken in indices)
+
+                    var indices = model.IncludeAll ? (await luceneIndexSettingsService.GetSettingsAsync()).Select(x => x.IndexName).ToArray() : model.Indices;
+
+                    foreach (var indexName in indices)
                     {
-                        var indexName = indexToken.ToObject<string>();
                         var luceneIndexSettings = await luceneIndexSettingsService.GetSettingsAsync(indexName);
                         if (luceneIndexSettings != null)
                         {
                             await luceneIndexingService.RebuildIndexAsync(indexName);
                             await luceneIndexingService.ProcessContentItemsAsync(indexName);
                         }
                     }
-                });
+                }, true);
```

Comment:
Why `bypassStateCheck` here and in `LuceneIndexResetStep`? Why is `bypassStateCheck` 
 necessary in the first place?

Reply:
It is necessary because of this **line** from **OrchardCore.Recipes** module:

![image](https://user-images.githubusercontent.com/6403130/189546757-cc47ee80-c4db-4227-9581-dd4eb0375b70.png)

as you can see before recipe executor takes it's action, we are setting shell state to **Initializing** so all subsequent requests respond with 503 (service unavailable)

Reply:
Maybe we could just not set the shellSettings state to Initializing in the Recipes controller, but potentialy a recipe step can be run during a setup which set the state to Initializing.

Maybe not block a backgroundJob if Initializing, we check this to prevent a job to be executed after a setup that may have failed. Hmm maybe not block at the beginning, then after the http request we could reload the settings to check if the state is still not running, I will look at it and let you know.

remind me if i forget ;)

Reply:
Hmm, I'd probably remove that `TenantState` check in `HttpBackgroundJob` altogether. Either check in the job if you need it, or check when queuing the job, but it's not necessarily generally a valid approach.

Awaiting what you think in the end JT, then.

Reply:
I updated `HttpBackgroundJob` here #12375 so that we also allow to trigger a job even if Initializing e.g. during a setup, then we check later on before the job that the tenant is running (like in a background task), e.g. if Uninitialized would mean that the setup failed, if still initializing would not be normal, at least after #12361, currently would also mean that the setup failed.

@lampersky can you give it a try, without having to use `bypassStateCheck`.

Reply:
@jtkech I've just tested it with your changes from #12375 and everything works!